glad/glad.go

```diff
@@ -136,12 +139,19 @@ func (u Updater) walkDir(root string) error {
 
 func (u Updater) searchPrefix(adv advisory, advisories []advisory) string {
 	for _, a := range advisories {
-		if a.PackageSlug == adv.PackageSlug {
-			continue
-		}
-
 		if strings.HasPrefix(adv.PackageSlug, a.PackageSlug) {
-			return a.PackageSlug
+			// we should not trim part of package name:
+			// e.g.: go/github.com/apache/thrift/lib/go/thrift and go/github.com/apache/thrift
+			// => go/github.com/apache/thrift/lib/go/thrift
+			// go/github.com/apache/thrift/lib/go/thrift and go/github.com/apache/thrift-mini
+			// => skip this advisory
+			suffix := strings.TrimPrefix(adv.PackageSlug, strings.TrimSuffix(a.PackageSlug, "/"))
+			if suffix == "/" { // same slugs
+				continue
+			}
+			if strings.HasPrefix(suffix, "/") {
+				return a.PackageSlug
+			}
```

Comment:
What if just adding `/` at last?

```
slug := adv.PackageSlug + "/"
```

Or

```
if !strings.HasSuffix(adv.PackageSlug, "/") {
    slug := adv.PackageSlug + "/"
}
```


Then, `go/github.com/apache/thrift/lib/go/thrift` and `go/github.com/apache/thrift-mini` will not match.

Reply:
I used regular expression. I think it is easier to understand. Can you take a look?

Reply:
Looks like `HasPrefix` is better. Any reason regexp is preferred?

Reply:
I thought regexp would be easier to understand, but i updated PR without regexp and followed your recommendations. Looks better. Thanks a lot!